src/main/java/com/redhat/devtools/lsp4ij/features/documentSymbol/LSPDocumentSymbolStructureViewFactory.java

```diff
@@ -0,0 +1,44 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at https://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * CppCXY
+ ******************************************************************************/
+package com.redhat.devtools.lsp4ij.features.documentSymbol;
+
+import com.intellij.ide.structureView.StructureViewBuilder;
+import com.intellij.ide.structureView.StructureViewModel;
+import com.intellij.ide.structureView.TreeBasedStructureViewBuilder;
+import com.intellij.lang.PsiStructureViewFactory;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.psi.PsiFile;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * LSP document symbol structure view factory.
+ */
+public class LSPDocumentSymbolStructureViewFactory implements PsiStructureViewFactory {
+
+    @Override
+    public @Nullable StructureViewBuilder getStructureViewBuilder(@NotNull PsiFile psiFile) {
+        return new LSPDocumentSymbolStructureViewBuilder(psiFile);
```

Comment:
We need really take care of this LSP instanciation since we should never override existing StructureViewBuilder (we could override it with API, but let'start with this idea).

At first you need to check if the file is associated to a language server. So please add on the first line of  the method:

```java
if (!LanguageServersRegistry.getInstance().isFileSupported(psiFile)) {
            return null;
}
```

But it is not enough, we need to find if it exists a builder for teh given psi file. The usecase is for our IJ Quarkus, we have a language server which is associated to a Java language but LSP must not override the Java builder.

Reply:
Replace with (hope it will work for you)

```java
@Override
    public @Nullable StructureViewBuilder getStructureViewBuilder(@NotNull PsiFile psiFile) {
        if (!LanguageServersRegistry.getInstance().isFileSupported(psiFile)) {
            // the file is not associated to a language server
            return null;
        }
        if (LanguageStructureViewBuilder.INSTANCE.allForLanguage(psiFile.getLanguage()).size() > 1) {
            // A plugin already contributes with custom PsiStructureViewFactory, don't override it.
            return null;
        }
        return new LSPDocumentSymbolStructureViewBuilder(psiFile);
    }
```